packages/@lwc/babel-plugin-component/src/utils.js

```diff
@@ -4,10 +4,27 @@
  * SPDX-License-Identifier: MIT
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
  */
-const { LWC_PACKAGE_ALIAS, LWC_PACKAGE_EXPORTS } = require('./constants');
+const moduleImports = require('@babel/helper-module-imports');
 const { LWCClassErrors, generateErrorMessage } = require('@lwc/errors');
 const lineColumn = require('line-column');
 
+const { LWC_PACKAGE_ALIAS, LWC_PACKAGE_EXPORTS } = require('./constants');
+
+/**
+ * Helper utility to work around babel regression introduced in
+ * https://github.com/babel/babel/pull/12331 where the opts for the current
+ * traversal is replaced with that of parent traversal. Specifically speaking,
+ * this issue was causing the Class() visitor to be clobbered after the first
+ * injection of registerDecorators(). Should be able to remove when
+ * https://github.com/babel/babel/issues/12570 is fixed.
+ */
+function addNamedImport(path, name, importedSource, opts) {
+    const cachedOpts = path.opts;
+    const specifierName = moduleImports.addNamed(path, name, importedSource, opts);
+    path.opts = cachedOpts;
+    return specifierName;
+}
```

Comment:
This is the only workaround I was able to figure out for the regression introduced for transforms that invoke a `traverse()` in a visitor.

Reply:
I'll investigate this to see if it fixes the issue...

Reply:
After debugging this with PM, we found that we still hit the regression. We are opting to remove the sub-traverse call instead.